simplified-cardcreator/src/main/java/org/nypl/simplified/cardcreator/ui/AccountInformationFragment.kt

```diff
@@ -0,0 +1,124 @@
+package org.nypl.simplified.cardcreator.ui
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.os.Bundle
+import android.text.Editable
+import android.text.TextWatcher
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.Observer
+import androidx.navigation.NavController
+import androidx.navigation.NavDirections
+import androidx.navigation.Navigation
+import org.nypl.simplified.cardcreator.R
+import org.nypl.simplified.cardcreator.databinding.FragmentAccountInformationBinding
+import org.nypl.simplified.cardcreator.utils.Cache
+import org.nypl.simplified.cardcreator.utils.hideKeyboard
+import org.nypl.simplified.cardcreator.viewmodels.UsernameViewModel
+import org.slf4j.LoggerFactory
+
+class AccountInformationFragment : Fragment() {
+
+  private val logger = LoggerFactory.getLogger(AccountInformationFragment::class.java)
+
+  private var _binding: FragmentAccountInformationBinding? = null
+  private val binding get() = _binding!!
+
+  private lateinit var navController: NavController
+  private lateinit var nextAction: NavDirections
+
+  private lateinit var sharedPreferences: SharedPreferences
+
+  private val minPinChars = 4
+  private val usernameMinChars = 5
+  private val usernameMaxChars = 25
+  private val usernameAvailable = "available-username"
+
+  private val viewModel: UsernameViewModel by viewModels()
+
+  override fun onCreateView(
+    inflater: LayoutInflater,
+    container: ViewGroup?,
+    savedInstanceState: Bundle?
+  ): View? {
+    _binding = FragmentAccountInformationBinding.inflate(inflater, container, false)
+    return binding.root
+  }
+
+  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+    super.onViewCreated(view, savedInstanceState)
+
+    navController = Navigation.findNavController(activity!!, R.id.card_creator_nav_host_fragment)
+
+    sharedPreferences = activity!!.getSharedPreferences(Cache.DEFAULT_PREFERENCE_NAME, Context.MODE_PRIVATE)
```

Comment:
Prefer `requireActivity()` and `requireContext()` to `activity!!`.

Reply:
You can also do something like this to get a locally scoped non-null instance:

```kotlin
val activity = requireActivity()
```

Under the hood it's basically doing:

```kotlin
val activity = this.activity ?: throw IllegalStateException()
```